Objects/typeobject.c

```diff
@@ -8795,8 +8823,9 @@ wrap_indexargfunc(PyObject *self, PyObject *args, void *wrapped)
     PyObject* o;
     Py_ssize_t i;
 
-    if (!PyArg_UnpackTuple(args, "", 1, 1, &o))
+    if (!check_num_args(args, 1))
```

Comment:
With this approach the error message won't include `__mul__` or `__rmul__`.

Maybe it's worth creating a `wrap_indexargfunc_r` just for `__rmul__` so the error messages can consistently include the right method names?

Reply:
It is okay, wrappers for other bina ry operators do not contain the dunder name.